s3-lambda-comprehend-cdk-python/README.md

```diff
@@ -0,0 +1,127 @@
+# S3 -> Lambda -> Comprehend
+This patterns shows CDK deployment on how to leverage Amazon S3, AWS Lambda, Amazon Comprehend and Amazon DynamoDB to perform Batch Sentiment Analysis in a serverless fashion.
+
+## Architecture
+![Diagram](src/architecture.jpg)
+
+### What resources will be created?
+This CDK code will create the following:
+   - 1 S3 bucket (to hold the text for which sentiment analysis has to be done)
+   - 1 Lambda (to invoke the Comprehend API)
+   - 1 DynamoDB (to store the result of Sentiment Analysis)
+   - 1 IAM role (for the Lambda to invoke Comprehend and DynamoDB)
+
+## Requirements
+
+### Development Environment
+**Cloud 9**
+
+This code for this pattern is executed in AWS Cloud9 environment. The EC2 instance used is m5.large (8 GiB RAM + 2 vCPU).
+
+### AWS setup
+**Region**
+
+If you have not yet run `aws configure` and set a default region, you must do so, or you can also run `export AWS_DEFAULT_REGION=<your-region>`. The region used in the demonstration is us-east-1 as Bedrock service is available only in limited regions.
+
+**Authorization**
+
+You must use a role that has sufficient permissions to create Iam roles, as well as cloudformation resources
+
+#### Python >=3.8
+Make sure you have [python3](https://www.python.org/downloads/) installed at a version >=3.8.x in the CDK environment. The demonstration has used python version 3.10.
+
+#### AWS CDK
+Make sure you have the [AWS CDK](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_install) installed in the Cloud9 environment.
+
+
+## Setup
+
+### Set up environment and gather packages
+
+```
+cd s3-lambda-comprehend-cdk-python
+```
+
+Install the required dependencies (aws-cdk-lib and constructs) into your Python environment 
+```
+pip install -r requirements.txt
+```
+
+### Gather and deploy resources with the CDK
+
+First synthesize, which executes the application, defines which resources will be created, and translates this into a cloudformation template
```

Comment:
What do you mean by "executes the application"?

Reply:
The intention was to help under what the `cdk synth` command does here. 
I've tried to summarize the contents from the AWS documentation here. 

https://docs.aws.amazon.com/cdk/v2/guide/hello_world.html#hello_world_tutorial_synth